express-notes-api/index.js

```diff
@@ -0,0 +1,115 @@
+const express = require("express");
+const app = express();
+const fs = require("fs");
+const notesJson = require("./data.json");
+
+const writingFile = () => {};
+
+app.use(express.json());
+
+app.get("/api/notes", (req, res) => {
+  const gradesArray = [];
+  if (notesJson.notes) {
+    for (let notes in notesJson.notes) {
+      gradesArray.push(notesJson.notes[notes]);
+    }
+  } else if (!notesJson.notes) {
+    res.json([]);
+  }
+  res.json(gradesArray);
+});
+
+app.get("/api/notes/:id", (req, res) => {
+  let noteGet = {};
+  let id = Math.trunc(+req.params.id);
+
+  for (let notes in notesJson.notes) {
+    if (notesJson.notes[id]) {
+      noteGet = notesJson.notes[id];
+      res.status(200);
+    } else if (id < 0 || isNaN(id)) {
+      noteGet.error = "Id must be a positive integer";
+      res.status(400);
```

Comment:
@GustavoAyala1 You're missing the point. The server should respond with a `400 Bad Request` if there's anything wrong with the format of the `id` (like it being a decimal number), not try to fix it for the client.

More importantly, are you testing your own work to make sure it produces the right outcomes?

---

![image](https://user-images.githubusercontent.com/16143771/144691678-3725102e-42fe-49d1-b167-b4fb97ba05a3.png)